cpp/Platform.Equality/Platform.Equality.Any.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#ifndef PLATFORM_EQUALITY_ANY_H
+#define PLATFORM_EQUALITY_ANY_H
+
+#include <any>
+#include <functional>
+#include <typeindex>
+#include <concepts>
+#include <stdexcept>
+
+namespace Platform::Equality
+{
+    template<class T>
+    inline auto __to_any_equal_visitor(auto func)
```

Comment:
Please rename it to `ToAnyEqualityComparer`. I think there is no need to add `__` at the start, if the user want to use it should be able to use it.

Reply:
It looks like it shouldn't be used by the user, the user should use RegisterEqualityComparer. Underscore notation is reserved by the standard https://en.cppreference.com/w/cpp/keyword 
>Also, all identifiers that contain a double underscore __ in any position and each identifier that begins with an underscore followed by an uppercase letter is always reserved and all identifiers that begin with an underscore are reserved for use as names in the global namespace. See identifiers for more details.

Therefore, to denote the "privacy" of the function, I propose to do the following way
```cpp
namespace Platform::Equality
{
namespace details 
{

template<class T>
inline auto ToAnyEqualityComparer(auto func) { /*...*/ }

}  // namespace details
}  // namespace Platform::Equality
```

Reply:
OK, we can either use `__ToAnyEqualityComparer` or use namespace `Internal` instead of `details`.